codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,286 @@
+ * predefined, cached instances where appropriate. Note that some results are {@link io.netty.util.ReferenceCounted}
+ * which are never cached/shared.
+ */
+public final class RedisMessageFactory {
+
+    private static PreloadStringsCache PRELOAD_SIMPLE_STRING;
+    private static PreloadStringsCache PRELOAD_ERRORS;
+
+    private static final ConstantBulkStringRedisMessage NULL_BULK_STRING = new ConstantBulkStringRedisMessage(null);
+
+    private static final ConstantBulkStringRedisMessage EMPTY_BULK_STRING =
+                                                new ConstantBulkStringRedisMessage(Unpooled.EMPTY_BUFFER);
+
+    private static final ConstantArrayRedisMessage NULL_ARRAY = new ConstantArrayRedisMessage(null);
+
+    private static final ConstantArrayRedisMessage EMPTY_ARRAY =
+                                                new ConstantArrayRedisMessage(Collections.<RedisMessage>emptyList());
+
+    private static final String[] DEFAULT_PRELOAD_SIMPLE_STRINGS = { "OK", "PONG", "QUEUED" };
+
+    private static final String[] DEFAULT_PRELOAD_ERRORS = {
+            "ERR",
+            "ERR index out of range",
+            "ERR no such key",
+            "ERR source and destination objects are the same",
+            "ERR syntax error",
+            "BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.",
+            "BUSYKEY Target key name already exists.",
+            "EXECABORT Transaction discarded because of previous errors.",
+            "LOADING Redis is loading the dataset in memory",
+            "MASTERDOWN Link with MASTER is down and slave-serve-stale-data is set to 'no'.",
+            "MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. " +
+                    "Commands that may modify the data set are disabled. Please check Redis logs for details " +
+                    "about the error.",
+            "NOAUTH Authentication required.",
+            "NOREPLICAS Not enough good slaves to write.",
+            "NOSCRIPT No matching script. Please use EVAL.",
+            "OOM command not allowed when used memory > 'maxmemory'.",
+            "READONLY You can't write against a read only slave.",
+            "WRONGTYPE Operation against a key holding the wrong kind of value",
+    };
+
+    static {
+        setPreloadStrings(RedisMessageType.SIMPLE_STRING, DEFAULT_PRELOAD_SIMPLE_STRINGS);
+        setPreloadStrings(RedisMessageType.ERROR, DEFAULT_PRELOAD_ERRORS);
+    }
+
+    private RedisMessageFactory() {
+    }
+
+    /**
+     * Set preload strings. Users can override default strings.
+     *
+     * @param type one of {@link RedisMessageType#SIMPLE_STRING} or {@link RedisMessageType#ERROR}.
+     * @param messages an array of String to preload.
+     */
+    public static void setPreloadStrings(RedisMessageType type, String[] messages) {
+        ObjectUtil.checkNotNull(messages, "messages must not be null");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)